third_party/docfx/templates/devsite/common.js

```diff
@@ -107,6 +108,30 @@ function getTitleForTypeProperty(type, name, uid) {
     }
 }
 
+/**
+ * Updates summary with an intro sentence containing the friendly API name.
+ *
+ * @param {string} summary The summary for the item.
+ * @param {string} type The type for the item (i.e. class, module).
+ * @param {string} friendlyApiName The friendly API title for the item.
+ * @param {Array} name The names for the item, only the first item is used for forming the sentence.
+ */
+function addIntroSentenceToSummary(summary, friendlyApiName, type, name) {

```

Comment:
This looks fairly complicated compared to adding it to the mustache template. How about doing this in the template instead with something like...

```
{{#summary}}
{{#friendlyApiName}} Reference documentation and code samples for the {{friendlyApiName}} {{title}}.{{/friendlyApiName}}
{{summary}}
{{/summary}
```

Reply:
It gets complicated when including the object's type in the sentence. I factored out the type addition to title so we wouldn't have to have a partial file for that. This would be similar: we'd need to have a partial for switching based on type. 

In general, I think we should try to avoid hard-coding language in the template -- it makes things like language updates and localization efforts more difficult.

Reply:
We won't be doing localization based on the template given here. That's fine. 

Another consideration is to add this for now into the Ruby YAMLs to begin with, which would be easier to find all the values rather than trying to compute and figure out what we need to extract from the YAMLs.

Reply:
Discussed offline, in summary:
There are trade-offs between at generator level or template level. There may be more complexity to implement this at the language-generator level, but it'd also simplify our template. We can revisit this approach if we expand this update to other languages.

Reply:
Not sure I understand. Feels funny to be dealing with `<p>` tags here when we have the HTML template, which should handle... the HTML? If we want to come up with the content here to avoid the type complexity, we should at least leave off the `<p>` and `</p>` and add those in the template.

Reply:
🤦🏻‍♂️, right that'd be a good idea! I put the <p> tags in the template, let me know what you think.